src/easylimit/rate_limiter.py

```diff
@@ -44,24 +45,56 @@ class RateLimiter:
     """
 
     def __init__(
-        self, max_calls_per_second: float = 1.0, track_calls: bool = False, history_window_seconds: int = 3600
+        self,
+        max_calls_per_second: Optional[float] = None,
+        rate_limit_calls: Optional[int] = None,
+        rate_limit_period: Optional[timedelta] = None,
+        track_calls: bool = False,
+        history_window_seconds: int = 3600,
     ) -> None:
         """
         Initialise the rate limiter.
 
         Args:
-            max_calls_per_second: Maximum number of calls allowed per second
+            max_calls_per_second: Maximum number of calls allowed per second (legacy parameter)
```

Comment:
Would it be possible to express the deprecation of this argument more overtly, e.g. but using method overloading with the old call signature annotated as deprecated?